src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleShell.cs

```diff
@@ -30,21 +31,23 @@ public static int Start(string bannerText, string helpText, string[] args)
         /// <param name="helpText">Help text for the shell.</param>
         /// <param name="args">Commandline parameters specified by user.</param>
         /// <returns>An integer value which should be used as exit code for the process.</returns>
-        public static int Start(InitialSessionState initialSessionState, string bannerText, string helpText, string[] args)
+        public static int Start(InitialSessionState initialSessionState, string? bannerText, string? helpText, string?[] args)
         {
-            if (initialSessionState == null)
+            if (args == null)
             {
-                throw PSTraceSource.NewArgumentNullException(nameof(initialSessionState));
+                throw PSTraceSource.NewArgumentNullException(nameof(args));
             }
```

Comment:
minor: Please move the validation on `args` back to after the `initialSessionState` validation, just like before your change.

Reply:
Done.